plugins/vst_base/VstPlugin.cpp

```diff
@@ -129,12 +132,20 @@ VstPlugin::VstPlugin( const QString & _plugin ) :
 	setSplittedChannels( true );
 
 	PE::MachineType machineType;
-	try {
-		PE::FileInfo peInfo(m_plugin);
-		machineType = peInfo.machineType();
-	} catch (std::runtime_error& e) {
-		qCritical() << "Error while determining PE file's machine type: " << e.what();
-		machineType = PE::MachineType::unknown;
+	QFileInfo fi(m_plugin);
+#ifdef LMMS_BUILD_LINUX
+	if (fi.completeSuffix() == "so")
+		machineType =PE::MachineType::linux64;
```

Comment:
PE means Portable Executable format for Windows binaries such as `.dll`, `.exe`. Having `linux` in `PE` namespace doesn't sound good, but it's not a big issue until this pull request is ready to merge.

Reply:
Yeah, I forgot about it, it was a hack done at the beginning to quickly see if it works :) Maybe should be changed to  `ExecutableFormat`?

Reply:
Let's leave this for now.
(Don't resolve this conversation now!)

Reply:
The `PE::MachineType` enum corresponds to these values here: https://docs.microsoft.com/en-gb/windows/win32/debug/pe-format#machine-types. I think it should be kept for the purpose of reading PE files, and we should introduce a separate enum to indicate which remote plugin host should be used.

For example:
```cpp
enum class ExecutableFormat
{
	Unknown, Win32, Win64, Linux64, // Maybe one day Linux32, Mac64, MacArm, etc.
};

// ...

auto format = ExecutableFormat::Unknown;

// ...

switch (peInfo.machineType())
{
	case PE::MachineType::amd64: format = ExecutableFormat::Win64;
	case PE::MachineType::i386: format = ExecutableFormat::Win32;
}

// ...

if (fi.suffix() == "so") { format = ExecutableFormat::Linux64; }
```